src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -308,6 +319,11 @@ export class MockHttpSocket extends MockSocket {
     serverResponse.removeHeader('connection')
     serverResponse.removeHeader('date')
 
+    // If the developer destroy the socket, gracefully destroy the response.
+    this.once('error', () => {
```

Comment:
The same for the mocked responses: if the socket gets destroyed, abort the pending server response.